Source/JavaScriptCore/dfg/DFGPlan.cpp

```diff
@@ -554,8 +554,9 @@ CompilationResult Plan::finalize()
             m_codeBlock->shrinkToFit(locker, CodeBlock::ShrinkMode::LateShrink);
         }
 
-        // Since Plan::reallyAdd could fire watchpoints (see ArrayBufferViewWatchpointAdaptor::add), it is possible that the current CodeBlock is now invalidated.
-        if (!m_codeBlock->jitCode()->dfgCommon()->isStillValid) {
+        // Since Plan::reallyAdd could fire watchpoints (see ArrayBufferViewWatchpointAdaptor::add),
+        // it is possible that the current CodeBlock is now invalidated & jettisoned.
+        if (m_codeBlock->isJettisoned()) {
```

Comment:
I still kinda like using the JIT code's isInvalidated here, since theoretically, we can reinstall a code block, which clears the isJettisoned bit. But of course, we should never actually reinstall a DFG code block that gets invalidated. Maybe assert the bits are equal?

Reply:
Discussed with Saam. JITCode's flag for unlinked code is always true, so we need to use this jettisoning flag. Since DFG code will not be re-installed into the same CodeBlock, this is fine.